4_params_munge/src/munge_contours.R

```diff
@@ -92,6 +92,84 @@ munge_ndgf_bathy <- function(out_ind, ndgf_contour_ind, ndgf_xwalk_ind, ndgf_sur
+                "nhdhr_{b6e2754f-7918-48bb-b6c5-e426895ad360}", "nhdhr_78542897", "nhdhr_155351290", "nhdhr_155641189", "nhdhr_04549854-A785-4DB7-9FC1-A3DEEBB23C49",
+                "nhdhr_155642807", "nhdhr_155642356","nhdhr_{1DDD22E2-1723-4447-B1A0-57E008ED2F28}")
+  # I did monotonicity tests on each remaining site_id group and these failed. I didn't get into trying to isolate one or more contours that were responsible,
+  # instead rejecting the whole lake and creating a new mega-issue as https://github.com/USGS-R/lake-temperature-model-prep/issues/167
+  failed_mono <- c('nhdhr_{13c737fd-517c-4f34-b813-79a31e833683}','nhdhr_{DF5F8AC1-4FD7-4D6D-9827-8692F742E74F}','nhdhr_120021837','nhdhr_123165517','nhdhr_145222228',
+                   'nhdhr_145222944','nhdhr_155351274','nhdhr_155471071','nhdhr_155472807','nhdhr_155472809','nhdhr_155641743','nhdhr_155641907','nhdhr_155642885',
+                   'nhdhr_49306509','nhdhr_53454933','nhdhr_56132527','nhdhr_62688605','nhdhr_96565971')
+
+  drop_ids <- c(bad_cast, failed_mono)
+
+  lake_surface <- sc_retrieve(lake_surface_ind) %>% readRDS() %>%
+    rename(geometry = Shape) %>% sf::st_zm() %>%
+    mutate(depths = 0) %>% dplyr::select(site_id, depths, geometry)
+
+  zip_file <- sc_retrieve(contour_zip_ind)
+
+  names_tbl <- purrr::map(1:length(LAKE_NAME_xwalk), function(j) {
+    tibble(LAKE_NAME = names(LAKE_NAME_xwalk)[j], site_id_a = LAKE_NAME_xwalk[j])
+    }) %>% purrr::reduce(bind_rows)
+
+  ids_tbl <- purrr::map(1:length(NHDID_WATE_xwalk), function(j) {
+    tibble(NHDID_WATE = names(NHDID_WATE_xwalk)[j], site_id_b = NHDID_WATE_xwalk[j])
+  }) %>% purrr::reduce(bind_rows)
+
+  # this is a simple internal function to avoid binding a bunch of new rows to the
+  # tibble since _most_ lakes in the `lake_surface` tbl are _not_ in this IN-specific dataset
+  # this also keeps us from having to filter out lake IDs that only have surface area in the
+  # hypso calculations
+  filter_id_and_bind <- function(existing_tbl, to_bind){
+    filtered_new <- filter(to_bind, site_id %in% unique(existing_tbl$site_id))
+    filtered_existing<- filter(existing_tbl, site_id %in% unique(filtered_new$site_id))
+    rbind(filtered_existing, filtered_new)
+  }
+
+  shp.path <- tempdir()
+  unzip(zip_file, exdir = shp.path)
+
+  #' still some bad geoms
+  #' requires sf version of at least 0.8:
+  sf::st_read(shp.path, layer = layer, stringsAsFactors=FALSE) %>%
+    sf::st_zm() %>%
+    mutate(site_id_c = paste0('nhdhr_', NHDID_WATE)) %>%
+    left_join(names_tbl, by = 'LAKE_NAME') %>%
+    left_join(ids_tbl, by = 'NHDID_WATE') %>%
+    # we want to order the columns so that dplyr::coalesce takes the first non-NA column
+    dplyr::select(site_id_a, site_id_b, site_id_c, everything()) %>%
+    mutate(site_id = coalesce(site_id_a, site_id_b, site_id_c)) %>%
+    dplyr::select(-site_id_a, -site_id_b, -site_id_c) %>%
+    # remove all sites that don't have any matches
```

Comment:
is this actually happening? I would expect to see a `filter(!is.na(site_id))` or something. Unless that is what `filter(site_id != 'nhdhr_NA')` is?

Reply:
I filtered after the mutate because some of the `is.na` actually get `site_ids` in the coalesce stage.